Log captured pic to firebase

Ayush-a3h opened the pull request.

SolbiatiAlessandro committed:
CAMERA - first commit, code structure

We copy the structure of the camera capture from https://github.com/theselfdrivingboat-com/AudioVideo/blob/2e238b9fb2160d767b835ab2c7e5721699c26cb5/CameraPreview/app/src/main/java/edu/cs4730/camerapreview/Camera2CapturePic.java
This code is untested! probably it will not work there are couple of
bugs.

@mack-4567 todo
- [ ] figure out that cameraId, where to take it what should it be
- [ ] add logging. You need to add log call in the code, and you need to
  write in the log what do you think the code is doing. Starting from
the very beginning of line 59 of SelfDrivingBoatCamera.java
- [ ] take a picture, figure out how to see the picture that was taken

SolbiatiAlessandro committed:
CAMERA - first commit, code structure

We copy the structure of the camera capture from https://github.com/theselfdrivingboat-com/AudioVideo/blob/2e238b9fb2160d767b835ab2c7e5721699c26cb5/CameraPreview/app/src/main/java/edu/cs4730/camerapreview/Camera2CapturePic.java
This code is untested! probably it will not work there are couple of
bugs.

@mack-4567 todo
- [ ] figure out that cameraId, where to take it what should it be
- [ ] add logging. You need to add log call in the code, and you need to
  write in the log what do you think the code is doing. Starting from
the very beginning of line 59 of SelfDrivingBoatCamera.java
- [ ] take a picture, figure out how to see the picture that was taken

SolbiatiAlessandro committed:
camera - getting images

getting them in form of bytes, now need to figure out how to send them
to AWS

SolbiatiAlessandro committed:
Merge branch 'camera' of https://github.com/theselfdrivingboat-com/BoatController into camera

Ayush-a3h committed:
[ADDED CODE TO UPLOAD PIC TO FIREBASE][WORKING]

-[X]Added basic code to upload captured pics to firebase

SolbiatiAlessandro commented on app/src/main/AndroidManifest.xml:
```diff
@@ -1,30 +1,25 @@
     <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION"/>
     <uses-permission android:name="android.permission.ACCESS_COARSE_LOCATION"/>
-
-    <uses-permission android:name="android.permission.INTERNET"/>
```
why we remove internet permission?

SolbiatiAlessandro commented on app/src/main/java/com/selfdrivingboat/boatcontroller/SelfDrivingBoatCamera.java:
```diff
@@ -116,9 +129,12 @@ public void onImageAvailable(ImageReader reader) {
                 output = new FileOutputStream(file);
                 output.write(bytes);
+                StorageReference reference = FirebaseStorage.getInstance().getReference().child(timeStamp);
+                UploadTask uploadTask = reference.putBytes(bytes);
```
nice this looks good to me! do you understand what bytes is? is basically the image in form of 0101010 if you print it you will get some HEX code (AE EF E9 E2) that is just a concise way to express binary.

SolbiatiAlessandro reviewed (commented):
LGTM, how do we get the pics in firebase

SolbiatiAlessandro commented:
Let's add steps to see images in firebase

SolbiatiAlessandro commented:
the last step we need for this, is to take a picture when we send the command. We can do it in this PR or in a new PR up to you @Mr-learner 

<img width="992" alt="Screenshot 2021-08-23 at 13 21 37" src="https://user-images.githubusercontent.com/20618047/130446245-11057591-f732-47ce-a350-f080518463db.png">
We are currently taking a picture when the app starts. That's good for testing but obviously we don't want that in production.

Here is the code that get called with heroku command https://github.com/theselfdrivingboat-com/BoatController/blob/dd1974abb30fcaaec41648b4ce950c8151f7268c/app/src/main/java/com/selfdrivingboat/boatcontroller/SelfDriving.java#L93

You can do something like this
- start self driving loop (i.e. test it with ESP32, you need the connection to start to get at that point of the code)
- go at "whereistheboat.com" and send a command clicking the big green button REQUEST IMAGE
- if you put the debugger at SelfDriving.java:93 you will se that command coming
- put a new switch case, if you get a REQUEST_IMAGE make the app take a pic and put it on firebase


Put a nice test plan! would be even better if you can record a short video where you show this full process end to end!  Like here I start the app, here I request the picture, here I get the picture on firebase. 

:) let me know if you have any question

Ayush-a3h committed:
[Added code to capture pic only on button call from site][Works but images are a little dark]

[<>]While runnig the app you might get a json file error,if in case you get the error,go to project setings in firebase console and download the json file.

[!]The pics are a bit dark,although i am not completely sure why its happening(to be checked)

[<>]While runnig the app,go to the website and press req img button a few times,and search for "pic request" and "pic captured" log,if they appear it means the pic was captured aftr the req img command was received,u might also get a few firebase authentication error,but check on the console first if the pic is received

Ayush-a3h committed:
Merge branch 'main' into CAM-firebase
